bazelisk.py

```diff
@@ -91,9 +91,25 @@ def determine_bazel_filename(version):
 
     return "bazel-{}-{}-{}".format(version, operating_system, machine)
 
+def determine_release_or_rc(version):
+    parts = version.lower().split("rc")
+    if len(parts) == 1:
+        # e.g. ("0.20.0", "release") for 0.20.0
+        return (version, "release")
+    elif len(parts) == 2:
+        # e.g. ("0.20.0", "rc2") for 0.20.0rc2
+        return (parts[0], "rc" + parts[1])
+    else:
+        raise Exception("Invalid version: {}. Versions must be in the form <x>.<y>.<z>[rc<rc-number>]".format(version))
+
+
 def download_bazel_into_directory(version, directory):
     bazel_filename = determine_bazel_filename(version)
-    url = "https://releases.bazel.build/{}/release/{}".format(version, bazel_filename)
+    (parsed_version, release_or_rc) = determine_release_or_rc(version)
```

Comment:
I think we could replace this with a regular expression, which would also correctly handle exceptions like the recent "0.20rc2" where the patch level version was accidentally missing:

```
(parsed_version, rc) = re.match(r'(\d*\.\d*(?:\.\d*)?)(rc\d)?', rc).groups()
```

Then use it inside a method that computes the URL:

```
def determine_url(version, bazel_filename):
  (parsed_version, rc) = re.match(r'(\d*\.\d*(?:\.\d*)?)(rc\d)?', rc).groups()
  return "https://releases.bazel.build/{}/{}/{}".format(parsed_version,
                                                        rc if rc else "release",
                                                        bazel_filename)
```